docs/sql-data-sources-csv.md

```diff
@@ -109,7 +109,7 @@ Data source options of CSV can be set via:
     <td>read</td>
   </tr>
   <tr>
-    <td><code>inferDate</code></td> 
+    <td><code>preferDate</code></td>
     <td>false</td>
     <td>Whether or not to infer columns that satisfy the <code>dateFormat</code> option as <code>Date</code>. Requires <code>inferSchema</code> to be <code>true</code>. When <code>false</code>, columns with dates will be inferred as <code>String</code> (or as <code>Timestamp</code> if it fits the <code>timestampFormat</code>).</td>
```

Comment:
```suggestion
    <td><code>preferDate</code></td>
    <td>false</td>
    <td>Whether or not to infer columns that satisfy the <code>dateFormat</code> option as <code>Date</code>. Inference requires <code>inferSchema</code> to be <code>true</code>. When <code>false</code>, columns with dates will be inferred as <code>String</code> (or as <code>Timestamp</code> if it fits the <code>timestampFormat</code>). When <code>true</code>, the parser will attempt entries in timestamp columns as DateType before other fallbacks</td>
```

What do you think of the following docs change to clarify the behavior? @cloud-fan @HyukjinKwon